src/utils/wallet.js

```diff
@@ -841,20 +843,43 @@ class Wallet {
             account.keyStore = tempKeyStore
             account.inMemorySigner = account.connection.signer = new nearApiJs.InMemorySigner(tempKeyStore)
             const newKeyPair = KeyPair.fromRandom('ed25519')
-
-            await this.addAccessKey(accountId, accountId, newKeyPair.publicKey, fromSeedPhraseRecovery)
-            if (i === accountIds.length - 1) {
-                await this.saveAndSelectAccount(accountId, newKeyPair)
-            } else {
-                await this.saveAccount(accountId, newKeyPair)
+            
+            try {
+                await this.addAccessKey(accountId, accountId, newKeyPair.publicKey, fromSeedPhraseRecovery)
+                accountIdsSuccess.push({
+                    accountId,
+                    newKeyPair
+                })
+            } catch (error) {
+                console.error(error)
+                accountIdsError.push({
+                    accountId,
+                    error
+                })
             }
         }))
 
         this.connection = connectionConstructor
 
-        return {
-            numberOfAccounts: accountIds.length,
-            accountList: accountIds.flatMap((accountId) => accountId.account_id).join(', '),
+        if (!!accountIdsSuccess.length) {
+            await Promise.all(accountIdsSuccess.map(async ({ accountId, newKeyPair }, i) => {
+                if (i === accountIdsSuccess.length - 1) {
```

Comment:
nit: this is not a blocker, but seems like logic can be somewhat simpler if we call `saveAccount` for every item in list and then call `selectAccount` for the last one after loop

Reply:
Sure. Additionaly refactored the same thing in `saveAndSelectLedgerAccounts` and small refactor to `saveAndSelectAccount`. 

I'm not sure about `setAccountConfirmed`, which is part of `saveAndSelectAccount` that was previously needed for confirming if account was ever loaded (createAccount error handling) - but I'm not sure if it's needed with the latest changes for this flow. I will add an issue for it.